nomad/state/schema.go

```diff
@@ -283,46 +283,32 @@ func jobVersionSchema() *memdb.TableSchema {
 
 // jobSubmissionSchema returns the memdb table schema of job submissions
 // which contain the original source material of each job, per version.
-// Unique index by Namespace, JobID, and Version.
 func jobSubmissionSchema() *memdb.TableSchema {
 	return &memdb.TableSchema{
 		Name: "job_submission",
 		Indexes: map[string]*memdb.IndexSchema{
 			"id": {
 				Name:         "id",
 				AllowMissing: false,
-				Unique:       true,
+				Unique:       true, // basically worthless
```

Comment:
```suggestion
				Unique:       true,
```

While arguably true, let's leave this off so we don't distract readers. AFAICT `Unique` always small optimizations in a handful of cases. It does not seem to enforce uniqueness, but I think your `uniqueness is DIY` comment below conveys that.